AdrianBarros/assignments/treesDFS/lc112/lc112.py

```diff
@@ -0,0 +1,70 @@
+# Problem Statement #
+
+# Given a binary tree and a number S, find if the tree has a path from root-to-leaf such that the sum of all the node values of that path equals S.
+
+
+class TreeNode:
+    def __init__(self, val, left=None, right=None):
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+'''
+input: a binary tree
+output: Bool that descrived if there is a sum or not
+
```

Comment:
Nice overview here - don't forget to mention the edge cases though.

Also make sure to analyze the time and space complexity after you come up with a solution.